### PR TITLE
Refactoring test_collections folder

### PR DESCRIPTION
**Pull request by rquidute**
### What changed
This PR is the phase-1 for the feature **Python tests in TH UI**. 
The motivation to refactoring is to prepare the code to retrieve the python tests from SDK. 
The final proposal for the folders organization is described bellow (PS: **python_testing** folder is part of phase 2):

+ test_collections
    + sample_tests
    + python_tests
    + sdk_tests	
      + sdk_checkout
         + .version
         + yaml_tests
            + yaml
        + python_testing
           + script
        + sdk_runner
      + support
        + yaml_tests
          + models
        + python_testing
           + models
        + __init__.py

### Related Issues
Python tests in TH UI - Phase 1 (Refactoring test_collection folder) - https://github.com/project-chip/certification-tool/issues/72

### Testing

-   Unit test updated and passing
Screenshot 2023-10-27 at 15 47 47

-   Test case execution successfully in TH  (Automated, Semi-Automated, Simulated, Manual)

### CODE REVIEW

**Review by mikaelhm — CHANGES_REQUESTED** [review]
I think it would be nice to keep a clear separation of content tracked in this repo and content pulled form the SDK repo.

I suggest that we organize it like this:

- `test_collections/sdk_tests/sdk_checkout/` is the only place we put content from SDK, and it only contains content from the SDK
- `test_collections/sdk_tests/support` could contain all the python code we need in our repo, for yaml models etc. But no code from the SDK repo.

This way we can simply delete `sdk_checkout` when we need to update.

File: .flake8 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [flake8]
 max-line-length = 88
 extend-ignore = E203, W503
-exclude = .git,__pycache__,__init__.py,.mypy_cache,.pytest_cache,third_party,test_collections/yaml_tests/yaml/sdk,test_collections/yaml_tests/sdk_runner
+exclude = .git,__pycache__,__init__.py,.mypy_cache,.pytest_cache,third_party,test_collections/sdk_tests/sdk_checkout/yaml_tests/yaml/sdk,test_collections/sdk_tests/sdk_checkout/sdk_runner
```
Comment (mikaelhm) [inline .flake8:4]: can we just exclude `test_collections/sdk_tests/sdk_checkout/`

File: .vscode/settings.json [hunk 2]
```diff
@@ -38,6 +38,6 @@
         "--no-cov"
     ],
     "python.linting.ignorePatterns": [
-        "test_collections/yaml_tests/yamls/**/*.py"
+        "test_collections/sdk_tests/sdk_checkout/yaml_tests/yaml/**/*.py"
```
Comment (mikaelhm) [inline .vscode/settings.json:41]: Again how about just ignoring 
```
test_collections/sdk_tests/sdk_checkout
```

File: cspell.json [hunk 3]
```diff
@@ -102,11 +102,11 @@
         "*.dockerfile",
         "*.pyc",
         ".vscode/*",
-        "test_collections/yaml_tests/sdk_runner",
-        "test_collections/yaml_tests/yaml/sdk",
+        "test_collections/sdk_tests/sdk_checkout/sdk_runner",
+        "test_collections/sdk_tests/sdk_checkout/yaml_tests/yaml/sdk",
```
Comment (mikaelhm) [inline cspell.json:106]: Again, just ignore `test_collections/sdk_tests/sdk_checkout/`

**Review by mikaelhm — APPROVED** [review]
Looks good.

But I think we can simplify the excluding of `test_collections/sdk_tests/sdk_checkout`

**Review by hiltonlima — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]

**Pull request merged**